Add new exercise in structural testing

OskarLorek opened the pull request:
This type of exercise is not yet included in the chapter. It demonstrates, using easy, example why for some expressions it is not possible to achieve MC/DC coverage and shows the consequences of this fact. It also teaches future-testers what to do in real-life situations when they encounter this type of expressions and why they should communicate it.

The exercise covers topics of MC/DC coverage and independence pairs.

The answer to the exercise with table is also included.

mauricioaniche commented:
Hi, @OskarLorek! There's an issue in this exercise! Tests 1, 2, and 3 would achieve 100% MC/DC!

I removed it from the book for now. Please, feel free to submit a corrected version!!

OskarLorek commented:
> Hi, @OskarLorek! There's an issue in this exercise! Tests 1, 2, and 3 would achieve 100% MC/DC!
> 
> I removed it from the book for now. Please, feel free to submit a corrected version!!

I fixed this exercise in #146

OskarLorek commented:
@mauricioaniche I understand why my mistake was. But I think that now there is also a mistake in MC/DC sub chapter of structural testing. ("It might not be possible to achieve MC/DC coverage in some expressions. See (A and B) or (A and not B)"). The argument for this in this paragraph is that there are no independence pairs of B, which it turns out does not imply impossibility of MC/DC coverage.
